File: 1833-find-the-highest-altitude/find-the-highest-altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function (gain) {
    let max = 0, ha = 0;
    for (let i = 0; i < gain.length; i++) {
        ha += gain[i];
        max = Math.max(max, ha);
    }
    return max;
};

File: 1833-find-the-highest-altitude/find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int max = 0, ha = 0;
        for(int i = 0; i< gain.length; i++){
            ha += gain[i];
            max = Math.max(max, ha);
        }
        return max;
    }
}